crates/sui-core/src/unit_tests/gas_tests.rs

```diff
@@ -241,6 +241,115 @@ async fn test_transfer_sui_insufficient_gas() {
     assert_eq!(effects.mutated()[0].1, sender);
 }
 
+/// - All gas coins should be owned by an address (not shared or immutable)
+/// - All gas coins should be owned by the sender, or the sponsor
+#[tokio::test]
+async fn test_invalid_gas_owners() {
```

Comment:
Can we also add tests for gases from both sponsor and sender? (reference: https://github.com/MystenLabs/sui/blob/main/crates/sui/tests/full_node_tests.rs#L141)

Reply:
@longbowlu I will do this, but want to put it in a separate PR to let this critical fix move forward ASAP.

Reply:
sound good